vspace/src/main/java/edu/asu/diging/vspace/web/general/search/PublicSearchSpaceController.java

```diff
@@ -0,0 +1,36 @@
+package edu.asu.diging.vspace.web.general.search;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.services.IPublicSearchManager;
+import edu.asu.diging.vspace.core.services.impl.model.SearchSpaceResults;
+
+@Controller
+public class PublicSearchSpaceController {
+
+    @Autowired
+    private IPublicSearchManager publicSearchManager;
+
+    @RequestMapping(value = "/exhibit/search/space")
+    public ResponseEntity<SearchSpaceResults> searchInVspace(
+            @RequestParam(value = "spacePagenum", required = false, defaultValue = "1") String spacePagenum,
+            Model model, @RequestParam(name = "searchText") String searchTerm) throws JsonProcessingException {
+
+        SearchSpaceResults publicSearch = publicSearchManager.searchForSpace(spacePagenum, searchTerm);      
```

Comment:
what's the different between `searchForSpace` and `searchSpaceAndPaginate`?

Reply:
Essentially, searchForSpace is called from PublicSearchSpaceController and searchSpaceAndPaginate is called from PublicSearchController.  Both these controllers have different use cases and require a different form of output. Hence these methods are kept different. Same applies to all other searches for modules/ slides/ slidetexts.

Reply:
I still don't get it, it seems like `SearchSpaceResults` has the same info as the `Page` that is returned from `searchForSpaceAndPaginate`, maybe page number is missing. Why not add missing information to the results object and then only have one method? It's very confusing right now.